### PR TITLE
Add a button Certificate in Dashboard (iCj) #316

### PR DESCRIPTION
**Pull request by straceX**
desired changes were made. used signal-slot mechanism

### CODE REVIEW

File: src/MainWindow.cpp [hunk 1]
```diff
@@ -382,11 +382,13 @@ void MainWindow::closeEvent(QCloseEvent *event)
         {
             saveSettings();
         }
+        emit closeEventSignal();
         QMainWindow::closeEvent(event);
     }
     else if (ret == QMessageBox::Discard)
     {
         saveSettings();
+        emit closeEventSignal();
```
Comment (xarkes) [inline src/MainWindow.cpp]: Don't use this, just do `new QDialog(this)` and it will automatically be closed.

**Comment by pelijah** [comment]
Use `git push --force` instead of opening new PR.

**Comment by straceX** [comment]
@xarkes  signal slot based solution changed QDialog based solution

File: src/widgets/Dashboard.cpp [hunk 2]
```diff
@@ -173,3 +178,19 @@ void Dashboard::updateContents()
     // Get stats for the graphs
     QStringList stats = CutterCore::getInstance()->getStats();
 }
+
+void Dashboard::on_certificateButton_clicked() {
+
+    if(certificateDialog->setJsonTreeView()) {
```
Comment (xarkes) [inline src/widgets/Dashboard.cpp]: Can you please indent the whole block correctly?

File: src/widgets/Dashboard.cpp [hunk 3]
```diff
@@ -12,14 +14,17 @@
...
+#include <QString>
+#include <QMessageBox>
+#include <QTreeView>
 
 Dashboard::Dashboard(MainWindow *main, QWidget *parent) :
     QDockWidget(parent),
     ui(new Ui::Dashboard),
-    main(main)
+    main(main),
+    certificateDialog(new JsonTreeViewDialog(this))
```
Comment (xarkes) [inline src/widgets/Dashboard.cpp]: Imho you can initialize the dialog when the button is clicked.

**Comment by xarkes** [comment]
Please rebase on top of master

**Comment by xarkes** [comment]
You don't have to close ... Just `push -f` to your branch.

**Pull request closed without merging**